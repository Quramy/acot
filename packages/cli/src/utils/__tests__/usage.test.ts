import stripAnsi from 'strip-ansi';
import { buildUsage } from '../usage';

describe('usage', () => {
  test('buildUsage - empty', () => {
    const output = buildUsage([]);

    expect(output).toBe('');
  });

  test('buildUsage - sections', () => {
    const output = buildUsage([
      {
        content: 'This is simple content.',
      },
      {
        content: 'Raw content\nOne\nTwo',
        raw: true,
      },
      {
        header: 'Title line',
        content: 'This is simple content.',
      },
      {
        content: [
          {
            name: '-f, --force',
            summary: 'Summary 1',
          },
          {
            name: '    --flag',
            summary: 'Summary 2',
          },
        ],
      },
    ]);

    expect(stripAnsi(output)).toMatchSnapshot();
  });
});
