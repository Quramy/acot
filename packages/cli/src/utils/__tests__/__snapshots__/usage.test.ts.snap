// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usage buildUsage - sections 1`] = `
"  This is simple content.

Raw content
One
Two

Title line
  This is simple content.

  -f, --force  Summary 1
      --flag   Summary 2"
`;
